controllers/front/AuthController.php

```diff
@@ -361,7 +361,7 @@ protected function processCustomerNewsletter(&$customer)
     {
         $blocknewsletter = Module::isInstalled('blocknewsletter') && $module_newsletter = Module::getInstanceByName('blocknewsletter');
         if ($blocknewsletter && $module_newsletter->active && !Tools::getValue('newsletter')) {
-            if (is_callable(array($module_newsletter, 'isNewsletterRegistered')) && $module_newsletter->isNewsletterRegistered(Tools::getValue('email')) == $module_newsletter::GUEST_REGISTERED) {
+            if (is_callable(array($module_newsletter, 'isNewsletterRegistered')) && $module_newsletter->isNewsletterRegistered(Tools::getValue('email')) == $module_newsletter->GUEST_REGISTERED) {
```

Comment:
yep but this " $module_newsletter::GUEST_REGISTERED" is providing the the "syntax error, unexpected T_PAAMAYIM_NEKUDOTAYIM"  becouse of "$module_newsletter" so your version is providing the 500 error page my is not elegant but will work until 7.0